addons/html_editor/static/src/others/embedded_components/core/video/video.js

```diff
@@ -15,18 +22,121 @@ export class EmbeddedVideoComponent extends Component {
+                target: this.videoBlock,
+                props: {
+                    overlay: this.videoSettingsOverlay,
+                    replaceVideo: () => {
+                        this.props.openVideoSelectorDialog((media) => {
+                            this.replaceVideo(media);
+                        }, iframe);
+                    },
+                    removeVideo: () => {
+                        this.videoBlock?.remove();
+                        this.videoSettingsOverlay.close();
+                        this.props.addStep();
+                        // After video removal, delay focus to ensure it's inside
+                        // the editable when the hint updates, avoiding incorrect
+                        // placeholder hints.
+                        setTimeout(() => this.props.focusEditable());
+                    },
+                },
+            });
+        });
+
+        useExternalListener(this.videoBlock, "mouseleave", (e) => {
+            if (e.relatedTarget?.closest(".video-overlay")) {
+                return;
+            }
+            this.videoSettingsOverlay.close();
+        });
+
+        onWillUnmount(() => {
+            this.videoSettingsOverlay?.close();
+        });
+    }
+
+    get url() {
+        return getVideoUrl(this.state.platform, this.state.videoId, this.state.params).toString();
+    }
+
+    /**
+     * Replace a video in the editor
+     * @param {Object} media
+     */
+    replaceVideo(media) {
+        this.state.videoId = media.videoId;
+        this.state.platform = media.platform;
+        this.state.params = media.params;
+        this.props.addStep();
+        this.props.focusEditable();
     }
 }
 
 export const videoEmbedding = {
     name: "video",
     Component: EmbeddedVideoComponent,
-    getProps: (host) => ({ ...getEmbeddedProps(host) }),
+    getProps: (host) => ({ host, ...getEmbeddedProps(host) }),
+    getStateChangeManager: (config) => new StateChangeManager(config),
 };
+
+export class VideoSettings extends Component {
+    static template = "html_editor.VideoSettings";
+    static components = { Dropdown, DropdownItem };
+    static props = {
+        replaceVideo: { type: Function },
+        removeVideo: { type: Function },
+        overlay: { type: Object },
+    };
+
+    setup() {
+        this.menuRef = useRef("menuRef");
+        this.dropdownState = useDropdownState();
+
+        onMounted(() => {
+            this.menuRef.el?.addEventListener("mouseleave", () => {
+                if (!this.dropdownState.isOpen) {
+                    this.props.overlay.close();
+                }
+            });
+        });
+
+        useExternalListener(document, "pointerdown", (ev) => {
+            if (ev.target.closest(".o-dropdown-item")) {
+                return;
+            }
+            this.props.overlay.close();
+        });
+
+        onWillUnmount(() => {
+            this.props.overlay.close();
+        });
```

Comment:
are there ways to unmount the VideoSettings component without closing the overlay? This feels redundant.